mmv1/products/chronicle/Watchlist.yaml

```diff
@@ -0,0 +1,127 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+name: Watchlist
+description: A watchlist is a list of entities that allows for bulk operations over the included entities.
+min_version: beta
+references:
+  guides:
+    'Google SecOps Guides': 'https://cloud.google.com/chronicle/docs/secops/secops-overview'
+  api: 'https://cloud.google.com/chronicle/docs/reference/rest/v1alpha/projects.locations.instances.watchlists'
+base_url: projects/{{project}}/locations/{{location}}/instances/{{instance}}/watchlists
+self_link: projects/{{project}}/locations/{{location}}/instances/{{instance}}/watchlists/{{watchlist_id}}
+create_url: projects/{{project}}/locations/{{location}}/instances/{{instance}}/watchlists?watchlistId={{watchlist_id}}
+id_format: projects/{{project}}/locations/{{location}}/instances/{{instance}}/watchlists/{{watchlist_id}}
+import_format:
+  - projects/{{project}}/locations/{{location}}/instances/{{instance}}/watchlists/{{watchlist_id}}
+update_verb: PATCH
+update_mask: true
+
+examples:
+  - name: 'chronicle_watchlist_basic'
+    primary_resource_id: 'example'
+    min_version: 'beta'
+    vars:
+      name: watchlist-name
+      description: watchlist-description
+    test_env_vars:
+      chronicle_id: 'CHRONICLE_ID'
+
+parameters:
+  - name: location
+    type: String
+    description: Resource ID segment making up resource `name`. It identifies the resource within its parent collection as described in https://google.aip.dev/122.
+    immutable: true
+    url_param_only: true
+    required: true
+  - name: instance
+    type: String
+    description: Resource ID segment making up resource `name`. It identifies the resource within its parent collection as described in https://google.aip.dev/122.
+    immutable: true
+    url_param_only: true
+    required: true
+  - name: watchlistId
+    type: String
+    description: |-
+      Optional. The ID to use for the watchlist,
+      which will become the final component of the watchlist's resource name.
+      This value should be 4-63 characters, and valid characters
+      are /a-z-/.
+    immutable: true
+    url_param_only: true
+    required: true
+properties:
+  - name: name
+    type: String
+    description: |-
+      Identifier. Resource name of the watchlist.
+      Format:
+      projects/{project}/locations/{location}/instances/{instance}/watchlists/{watchlist}
+    default_from_api: true
+  - name: multiplyingFactor
+    type: Double
+    description: |-
+      Optional. Weight applied to the risk score for entities
+      in this watchlist.
+      The default is 1.0 if it is not specified.
```

Comment:
Should add `default_from_api: true`?

Reply:
Is this because the field is an optional field and should update the value from the API response if not provided by the user? Should we update all optional fields to have this?

Reply:
When the field is not specified in the Terraform configuration, but a value is returned from API, then `default_from_api: true` is needed. I wonder if the default 1.0 is returned from API or not, if no value is specified in the configuration.

Reply:
We shouldn't update all optional fields to have this.

Reply:
> When the field is not specified in the Terraform configuration, but a value is returned from API, then default_from_api: true is needed. I wonder if the default 1.0 is returned from API or not, if no value is specified in the configuration.

I will double check on that and update accordingly. Thanks for the heads-up.

Reply:
So, multiplyingFactor is not returned from the API when we don't specify it. Hence, `default_from_api: true` isn't required here. Although, a couple of other properties had this behavior where the API returned them in response even if they where not specified in the config (`watchlistUserPreferences` and `manual ` property under `entityPopulationMechanism `). Hence, updated them to have `default_from_api: true`.

Kindly review.